### PR TITLE
doc(whl): add code doc for LT,DT,PC,BC models

### PR DESCRIPTION
**Pull request by kxzxvbk**
## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/model/template/bc.py [hunk 1]
```diff
@@ -173,14 +178,27 @@ def __init__(
                 )
             )
 
-    def forward(self, inputs: Union[torch.Tensor, Dict[str, torch.Tensor]]) -> Dict[str, torch.Tensor]:
+    def forward(self, inputs: Union[torch.Tensor, Dict[str, torch.Tensor]]) -> Dict:
```
Comment (PaParaZz1) [inline ding/model/template/bc.py:181]: why delete `Dict[str, torch.Tensor]`

File: ding/model/template/bc.py [hunk 2]
```diff
@@ -173,14 +178,27 @@ def __init__(
...
+            >>> model = ContinuousBC(32, 6, action_space='regression')
+            >>> inputs = torch.randn(4, 32)
+            >>> outputs = model(inputs)
+            >>> assert isinstance(outputs, dict) and outputs['action'].shape == torch.Size([4, 6])
+
+        Examples (Reparameterization):
+            >>> model = ContinuousBC(32, 6, action_space='reparameterization')
+            >>> inputs = torch.randn(4, 32)
+            >>> outputs = model(inputs)
+            >>> assert isinstance(outputs, dict) and outputs['logit'][0].shape == torch.Size([4, 6])\
```
Comment (PaParaZz1) [inline ding/model/template/bc.py:200]: use two lines assertions

File: ding/model/template/bc.py [hunk 3]
```diff
@@ -36,7 +41,7 @@ def __init__(
             - head_hidden_size (:obj:`Optional[int]`): The ``hidden_size`` of head network.
             - head_layer_num (:obj:`int`): The number of layers used in the head network to compute Q value output
             - activation (:obj:`Optional[nn.Module]`): The type of activation function in networks \
-                if ``None`` then default set it to ``nn.ReLU()``
+                if ``None`` then default set it to ``nn.ReLU()``.
```
Comment (PaParaZz1) [inline ding/model/template/bc.py:44]: don't miss any of arguments, such as `strides` here

File: ding/model/template/decision_transformer.py [hunk 4]
```diff
@@ -13,14 +13,32 @@
...
 import torch.nn.functional as F
+from ding.utils import SequenceType
 
 
 class MaskedCausalAttention(nn.Module):
-
-    def __init__(self, h_dim, max_T, n_heads, drop_p):
+    r"""
+    Overview:
+        The implementation of masked causal attention in decision transformer.
```
Comment (PaParaZz1) [inline ding/model/template/decision_transformer.py:27]: add more explanation about keypoints of this module, such as mask and causal

File: ding/model/template/decision_transformer.py [hunk 5]
```diff
@@ -95,20 +158,42 @@ def forward(self, x):
...
+            - obs_shape (:obj:`Union[int, SequenceType]`): Dimension of state, such as 128, (4, 84, 84).
+            - act_dim (:obj:`int`): The dimension of actions, such as 6.
+            - n_blocks (:obj:`int`): The number of transformer blocks in the decision transformer, such as 3.
+            - h_dim (:obj:`int`): The dimension of hidden states, such as 128.
+            - context_len (:obj:`int`): The max context length of the attention, such as 6.
+            - n_heads (:obj:`int`): The number of heads in calculating attention, such as 8.
+            - drop_p (:obj:`float`): The drop rate of the drop-out layer, such as 0.1.
+            - max_timestep (:obj:`int`): The max length of the total sequence, defaults to be 4096.
+            - state_encoder (:obj:`Optional[nn.Module]`): The encoder to pre-process the given input. If it is set to \
+            None, the raw state will be pushed into the transformer.
```
Comment (PaParaZz1) [inline ding/model/template/decision_transformer.py:194]: there needs a proper indent

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: ding/model/template/decision_transformer.py [hunk 6]
```diff
@@ -13,14 +13,32 @@
...
+    Interfaces:
+        ``__init__``, ``forward``
+    """
+
+    def __init__(self, h_dim: int, max_T: int, n_heads: int, drop_p: float) -> None:
+        """
+        Overview:
+            Initialize the MaskedCausalAttention Model according to input arguments.
+        Arguments:
+            - h_dim (:obj:`int`): The dimension of hidden states, such as 128.
```
Comment (AltmanD) [inline ding/model/template/decision_transformer.py:37]: May be more appropriate to adjust it to the ‘The dimension of the hidden layers’ here.

File: ding/model/template/decision_transformer.py [hunk 7]
```diff
@@ -70,8 +103,23 @@ def forward(self, x):
...
+    Interfaces:
+        ``__init__``, ``forward``
+    """
+
+    def __init__(self, h_dim: int, max_T: int, n_heads: int, drop_p: float) -> None:
+        """
+        Overview:
+            Initialize the Block Model according to input arguments.
+        Arguments:
+            - h_dim (:obj:`int`): The dimension of hidden states, such as 128.
```
Comment (AltmanD) [inline ding/model/template/decision_transformer.py:118]: same to line 37

File: ding/model/template/decision_transformer.py [hunk 8]
```diff
@@ -95,20 +158,42 @@ def forward(self, x):
...
+        continuous: bool = False
     ):
+        """
+        Overview:
+            Initialize the DecisionTransformer Model according to input arguments.
+        Arguments:
+            - obs_shape (:obj:`Union[int, SequenceType]`): Dimension of state, such as 128, (4, 84, 84).
+            - act_dim (:obj:`int`): The dimension of actions, such as 6.
+            - n_blocks (:obj:`int`): The number of transformer blocks in the decision transformer, such as 3.
+            - h_dim (:obj:`int`): The dimension of hidden states, such as 128.
```
Comment (AltmanD) [inline ding/model/template/decision_transformer.py:188]: same to line 37

File: ding/model/template/decision_transformer.py [hunk 9]
```diff
@@ -95,20 +158,42 @@ def forward(self, x):
...
+        drop_p: float,
+        max_timestep: int = 4096,
+        state_encoder: Optional[nn.Module] = None,
+        continuous: bool = False
     ):
+        """
+        Overview:
+            Initialize the DecisionTransformer Model according to input arguments.
+        Arguments:
+            - obs_shape (:obj:`Union[int, SequenceType]`): Dimension of state, such as 128, (4, 84, 84).
```
Comment (AltmanD) [inline ding/model/template/decision_transformer.py:185]: adjust 'such as 128, (4, 84, 84)' to 'such as 128 or (4, 84, 84)' is more appropriate.

File: ding/model/template/decision_transformer.py [hunk 10]
```diff
@@ -152,7 +237,60 @@ def __init__(
...
+        """
+        Overview:
+            Forward computation graph of the decision transformer, input a sequence tensor \
+            and return a tensor with the same shape.
+        Arguments:
+            - timesteps (:obj:`torch.Tensor`): The timestep for input sequence.
+            - states (:obj:`torch.Tensor`): The sequence of states.
+            - actions (:obj:`torch.Tensor`): The sequence of actions.
+            - returns_to_go (:obj:`torch.Tensor`): The sequence of return-to-go.
+            - tar (:obj:`Optional[int]`): The targe index.
```
Comment (AltmanD) [inline ding/model/template/decision_transformer.py:257]: this 'tar' is use to control whether to predict action, regardless of index.

File: ding/model/template/procedure_cloning.py [hunk 11]
```diff
@@ -9,12 +9,30 @@
...
+        Overview:
+            Initialize the procedure cloning transformer model according to corresponding input arguments.
+        Arguments:
+            - cnn_hidden (:obj:`int`): The last channel dimension of CNN encoder, such as 32.
+            - att_hidden (:obj:`int`): The dimension of attention blocks, such as 32.
+            - att_heads (:obj:`int`): The number of heads in attention blocks, such as 4.
+            - drop_p (:obj:`float`): The drop out rate of attention, such as 0.5.
+            - max_T (:obj:`int`): The sequence length of procedure cloning, such as 4.
+            - n_attn (:obj:`int`): The number of attention layers, such as 4.
+            - feedforward_hidden (:obj:`int`): The number of feedforward layers, such as 4.
```
Comment (AltmanD) [inline ding/model/template/procedure_cloning.py:34]: miss `n_feedforward`

File: ding/model/template/bc.py (hunk 1 above)
Comment (kxzxvbk) [reply ding/model/template/bc.py:181]: That's because I find that when the action space is "reparameterization", the returned type should be ``Dict[str, List]``, which is not compatible to the original doc.

**Review by PaParaZz1 — APPROVED** [review]
